hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/InvocationRegistry.java

```diff
@@ -106,7 +106,8 @@ long getLastCallId() {
     public boolean register(Invocation invocation) {
         final long callId;
         try {
-            callId = callIdSequence.next(invocation.op.isUrgent());
+            boolean force = invocation.op.isUrgent() || invocation.isRetryCandidate();
```

Comment:
@pveentjer: you're right. somehow we were thinking that register & deregister will be done by the same thread (InvocationMonitor.scheduler).
But deregister is (mostly) done by response thread. So, yes, registry can grow unbounded.